src/v2/Apps/Home/Components/HomeArtworkModuleRail.tsx

```diff
@@ -0,0 +1,239 @@
+
+const ARTWORK_MODULES = [
+  "current_fairs",
+  "followed_artist",
+  "generic_gene", // GENERIC_GENES
+  "genes", // FOLLOWED_GENES
+  "live_auctions",
+  "recently_viewed_works",
+  "recommended_works",
+  "related_artists",
+  "saved_works",
+  "similar_to_recently_viewed",
+  "similar_to_saved_works",
+] as const
+
+const UNIMPLEMENTED_MODULES = [
+  "active_bids",
+  "followed_artists",
+  "followed_galleries",
+  "popular_artists",
+] as const
+
+const CONTEXT_MODULES: Record<
+  typeof ARTWORK_MODULES[number],
+  AuthContextModule
+> = {
+  current_fairs: ContextModule.fairRail,
+  followed_artist: ContextModule.otherWorksByArtistRail,
+  generic_gene: ContextModule.categoryRail,
+  genes: ContextModule.categoryRail,
+  live_auctions: ContextModule.liveAuctionsRail,
+  recently_viewed_works: ContextModule.recentlyViewedRail,
+  related_artists: ContextModule.relatedArtistsRail,
+  // @ts-ignore TODO: Add to AuthContextModule union
+  recommended_works: ContextModule.recommendedWorksForYouRail,
+  // @ts-ignore TODO: Add to AuthContextModule union
+  saved_works: ContextModule.recentlySavedRail,
+  // @ts-ignore TODO: Add to AuthContextModule union
+  similar_to_recently_viewed: ContextModule.similarToWorksYouViewedRail,
+  // @ts-ignore TODO: Add to AuthContextModule union
+  similar_to_saved_works: ContextModule.similarToWorksYouSavedRail,
+}
+
+interface HomeArtworkModuleRailProps {
+  artworkModule: HomeArtworkModuleRail_artworkModule
+}
+
+const HomeArtworkModuleRail: React.FC<HomeArtworkModuleRailProps> = ({
+  artworkModule,
+}) => {
+  if (
+    UNIMPLEMENTED_MODULES.includes(
+      artworkModule.key as typeof UNIMPLEMENTED_MODULES[number]
+    )
+  ) {
+    console.warn(
+      `TODO: ${artworkModule.key}:${artworkModule.context?.__typename}`
+    )
+    return null
+  }
+
+  const artworks = compact(artworkModule.results)
+
+  if (artworks.length === 0) {
+    return null
+  }
+
+  return (
+    <HomeArtworkModule
+      context={
+        <HomeArtworkModuleContextFragmentContainer
+          title={artworkModule.title}
+          context={artworkModule.context}
+        />
+      }
+    >
+      <Shelf>
+        {artworks.map(artwork => {
+          return (
+            <ShelfArtworkFragmentContainer
+              contextModule={CONTEXT_MODULES[artworkModule.key!]}
+              artwork={artwork}
+              lazyLoad
+            />
+          )
+        })}
+      </Shelf>
+    </HomeArtworkModule>
+  )
+}
+
+const HomeArtworkModuleRailFragmentContainer = createFragmentContainer(
+  HomeArtworkModuleRail,
+  {
+    artworkModule: graphql`
+      fragment HomeArtworkModuleRail_artworkModule on HomePageArtworkModule {
+        title
+        key
+        results {
+          ...ShelfArtwork_artwork
+        }
+        context {
+          __typename
+          ...HomeArtworkModuleContext_context
+        }
+      }
+    `,
+  }
+)
+
+const HomeArtworkModule: React.FC<{ context: string | JSX.Element }> = ({
+  context,
```

Comment:
Thoughts on using `context` as a prop? This could get confusing later

Reply:
I suppose; the problem being of course that this is named `context` in the schema.

![](https://static.damonzucconi.com/_capture/sdHerowf.png)

Maybe just a clarifying comment?

Reply:
If its in the schema then its in the schema, no worries.